server/controllers/auth.js

```diff
@@ -52,3 +56,63 @@ exports.login = async (req, res, next) => {
     }
   }
 };
+
+exports.registerController = async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      password,
+      email,
+      role,
+      customized,
+    } = await registerSchema.validate(req.body, {
+      abortEarly: false,
+    });
+    let existingUser;
+    switch (role) {
+      case 'artist':
+        existingUser = await getArtistByEmail(email);
+        break;
+
+      case 'customer':
+        existingUser = await checkCustomerEmail(email);
+        break;
+      default:
+        throw new Error('Choose your role');
+    }
+    if (existingUser.rows.length !== 0)
+      return res
+        .status(400)
+        .json({ statusCode: 400, message: 'email exist !!' });
+
+    let newPassword;
+    try {
+      const salt = await genSalt(10);
+      const hashPass = await hash(password, salt);
+      newPassword = hashPass;
+    } catch (error) {
+      next(error);
+    }
+    const newUser = {
+      firstName,
+      lastName,
+      email,
+      password: newPassword,
+      customized,
+    };
```

Comment:
I think you don't need to use nested ` try/catch`, one `try/catch` for the whole method will be enough  to catch the error.
```suggestion  
    const salt = await genSalt(10);
    const hashPass = await hash(password, salt);
    const newUser = {
      firstName,
      lastName,
      email,
      password: hashPass,
      customized,
    };
```